BlinkID/example/lib/main.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:blinkid_flutter/microblink_scanner.dart';
+import 'package:flutter/material.dart';
+import 'package:sample/document_scanner.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const HomePage(),
+    );
+  }
+}
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('BlinkID Example')),
+      body: Center(
+        child: FilledButton(
+          onPressed: () async {
+            final result = await context.scanDocument();
+            if (!context.mounted || result == null) return;
+
+            showDialog(
+                context: context,
+                builder: (_) => Dialog(
+                      clipBehavior: Clip.antiAlias,
+                      child: Scaffold(
+                        appBar: AppBar(title: const Text('Result'), leading: const CloseButton()),
+                        body: ListView.builder(
+                          itemBuilder: (_, index) {
+                            final item = result[index];
+
+                            final name = switch (item) {
+                              BlinkIdMultiSideRecognizerResult _ => 'BlinkIdMultiSideRecognizerResult',
+                              _ => item.runtimeType.toString(),
+                            };
+
+                            return ExpansionTile(
+                                title: Text(name),
+                                children: item.formatted.entries.map((entry) {
+                                  return ListTile(title: Text('${entry.key}: ${entry.value}'));
+                                }).toList());
+                          },
+                          itemCount: result.length,
+                        ),
+                      ),
+                    ));
+          },
+          child: const Text('Scan Document'),
+        ),
+      ),
+    );
+  }
+}
+
+extension on RecognizerResult {
+  Map<String, dynamic> get formatted => switch (this) {
```

Comment:
Why? There is already the `toJson()` function defined on `RecognizerResult`. If this app is to understand the package, better not to introduce anything on top of it.

Reply:
Have you checked what the `toJson()`  returns? 😅

It's useless.

```json
{
    "resultState": 2
}
```

<img width="582" alt="image" src="https://github.com/user-attachments/assets/51dd438a-47c1-44aa-b9cd-31f0c95355e9">

Reply:
Update `BlinkIdMultiSideRecognizerResult` to store `Map<String, dynamic> nativeResult`.

Reply:
No. It's out of scope. I also don't want to change the types much because when I'm updating the library there are too many changes to fix.

Reply:
The goal is to gain proper insight into the `RecognizerResult`, making it appropriate to address this in the current PR. More so, if you care about not having a lot of problems when updating with the upstream repo. Cause keeping `formatted` as it is might seem beneficial, but it could lead to complications later, especially when updating with the upstream repo (new properties might be added, old - renamed, etc.). But if you simply get the "nativeResult" which is a JSON, and won't be asserting any structure of it - you wouldn't have those problems. The only "problem" really is the possibility of an easy to resolve conflict - if you add `nativeResult` as a property to `RecognizerResult`.

Reply:
Ok I didn't read the comment right. It's just about adding the native result as an argument.

Reply:
Done 👍🏻